trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/security/S3DatabaseSecurityController.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.server.security;
+
+import io.airlift.log.Logger;
+import io.trino.aws.proxy.server.rest.RequestLoggerController;
+import io.trino.aws.proxy.server.rest.RequestLoggingSession;
+import io.trino.aws.proxy.spi.rest.ParsedS3Request;
+import io.trino.aws.proxy.spi.security.S3DatabaseSecurityFacade;
+import io.trino.aws.proxy.spi.security.S3DatabaseSecurityFacadeProvider;
+import io.trino.aws.proxy.spi.security.S3SecurityFacade;
+import io.trino.aws.proxy.spi.security.S3SecurityFacadeProvider;
+import io.trino.aws.proxy.spi.security.SecurityResponse;
+import jakarta.ws.rs.WebApplicationException;
+
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+class S3DatabaseSecurityController
```

Comment:
any reason this class is not public?

Reply:
Why? It doesn't need to be.

Reply:
Fair point, I am in the habit of making all classes public in service code.

Reply:
In my experience if you make it public someone will use it and then you get stuck having to support it. But, of course, we can always make it public later.